gsoc/proposal-Implementation of ML Framework to Prioritize Meaningful Changes In Environment Related Websites.md

```diff
@@ -0,0 +1,280 @@
+
+**Name:** Ankur Shukla
+
+**Affiliation:** Pursuing Geoinformatics and Natural Resources Engineering, at Indian Institute of Technology, Bombay as Masters Degree
+
+**Location:** Mumbai, India, UTC +5:30
+
+**GitHub handle:** [daas-ankur-shukla](https://github.com/daas-ankur-shukla)
+
+**Slack handle:** daas-ankur
+
+**Email:** work.ankurshukla@gmail.com
+
+**Other Contact: **[ https://www.linkedin.com/in/ankur-shukla-8b73b266/](https://www.linkedin.com/in/ankur-shukla-8b73b266/)
+
+### **Project Details**
+
+**Project Title:**
+
+Implementation of ML Framework to prioritize meaningful changes in Environment related websites
+
+**Project Abstract/Summary:**
+
+The project aims to develop and implement a machine learning framework to prioritize diffs and generate appropriate actions.
+
+**Describe the need your project fills:**
+
+This project aims to fulfill the following aims to priorities the webpage diffs:
+
+1. Conversion of html diffs to a common schema for machine learning algorithms
+
+2. Machine Learning model to accept formated diffs as input and output their priority
+
+3. Integrate the model and other components with other EDGI components to take human analyst feedback for improving the model.
+
+**Describe how your project meets these needs:**
+
+A website can undergo changes at various levels, viz: its design, layout, content and interface. For this project we are concerned with changes in the content of a website. Further the content of a website can be categorized as images, text (headings, paragraphs). In this project I propose to deal with text changes in a website.
+
+To get the most out of machine learning models, we first have a definite diff format which can be fed to ML models. Thus the first focus of my project will be to arrive at a schema for converting various diff formats to a common format.
+
+1. Conversion of html diffs to a common schema to form the feature matrix for machine learning model
+
+After thorough discussion, we can arrive at a feature matrix, with the best possible feature specification to train our ML model. The following can be a prospective list of features (non-exhaustive):
+
+1. HTML Tag difference
+
+2. Word Difference
+
+3. Position Difference
+
+There shall be intermediate vectors for each class of change. For example for HTML tags the following may be a priority vector (non-exhaustive):
+
+<table>
+  <tr>
+    <td>Tag</td>
+    <td><h1></td>
+    <td><h2></td>
+    <td><b></td>
+    <td><p></td>
+  </tr>
+  <tr>
+    <td>Priority</td>
+    <td>4</td>
+    <td>3</td>
+    <td>2</td>
+    <td>1</td>
+  </tr>
+</table>
+
+
+These priority vectors can help us give a relative severity index to a change derived from a diff. For example, a change from <p> tag to <h1> is a more severe change as compared to a change from <b> to <h2>. There will be similar vectors for severity for different change types. These change severity vectors can then be given to a machine learning classifier to prioritize them
```

Comment:
Just a tip: the HTML tags like `<h1`> here need to be quoted because otherwise they are interpreted by GitHub's Markdown processor and screw up the formatting of the document. (View the document in the normal view to see what I mean.)